src/olympia/accounts/helpers.py

```diff
@@ -20,6 +47,42 @@ def fxa_config(context):
     return config
 
 
+def link_with_final_destination(context, base):
+    return urlparams(base, to=path_with_query(context['request']))
+
+
+def path_with_query(request):
+    next_path = request.path
+    qs = request.GET.urlencode()
+    if qs:
+        return '{next_path}?{qs}'.format(next_path=next_path, qs=qs)
```

Comment:
While you are at it, modify your tests to add some utf-8 chars in the test paths to make sure things still work ?